src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,235 @@
+package duke.task;
+
+import duke.Duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    /**
+     * Total number of task object in current list.
+     */
+    private int numOfTasks = 0;
+    /**
+     * The collection of task object in current list.
+     */
+    private ArrayList<Task> Tasks;
+
+    /**
+     * Initializes a to-do-list object with an empty array list that stores task item.
+     */
+    public TaskList() {
+        this.Tasks = new ArrayList<>();
+    }
+
+    /**
+     * Returns the total number of task object.
+     *
+     * @return the total number of task object.
+     */
+    public int getNumOfTasks() {
+        return numOfTasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return Tasks;
+    }
+
+    public void setTasks(ArrayList<Task> tasks) {
+        Tasks = tasks;
+    }
+
+    public void setNumOfTasks(int numOfTasks) {
+        this.numOfTasks = numOfTasks;
+    }
+
+    /**
+     * Returns a string message about the number of task objects in the list.
+     *
+     * @return a string of the number of task objects in the list.
+     */
+    private String getNumOfTasksString() {
+        int numOfTasks = getNumOfTasks();
+        return "Now you have " + numOfTasks + (numOfTasks == 1 ? " task" : " tasks") + " in the list.";
```

Comment:
Though I understand the intended function of this line, it's generally best to avoid conditionals in favour of a more explicitly clear if-else statement.